common/numeric-limits.h

```diff
@@ -0,0 +1,89 @@
+/* ScummVM - Graphic Adventure Engine
+ *
+ * ScummVM is the legal property of its developers, whose names
+ * are too numerous to list here. Please refer to the COPYRIGHT
+ * file distributed with this source distribution.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef COMMON_NUMERIC_LIMITS_H
+#define COMMON_NUMERIC_LIMITS_H
+
+#include "common/scummsys.h"
+#include "common/type-traits.h"
+
+namespace Common {
+
+template <typename T, bool TIsIntegral, bool TIsUnsigned>
+struct IntegralNumericLimits {
+};
+
+template<typename T>
+struct IntegralNumericLimits<T, true, false> {
+	static const T max_value = ((T(1) << (sizeof(T) * 8 - 2)) - 1) * 2 + 1;
+	static const T min_value = -max_value - 1;
+	static const T lowest_value = min_value;
```

Comment:
These constants should not be here.
Either private or not here at all to not expose them.
If we mimic C++11 STL, only functions are to be used.

Reply:
OK, I'll move those into something else... but I do think these values need to be exposed in a way that doesn't need to call a function.  It's not just numeric_limits equivalent that's missing, it's also things like INT_MAX and such... so while the functions will work for a lot of cases, they won't work for cases that either need a numeric constant (e.g. template args) or where using a function call would violate code standards by having an on-init function call.

Reply:
Maybe you can try constexpr?

Reply:
INT_MAX and so are really a different thing: they are macros handled by the preprocessor and can be used in `#if` directives.